addons/respawn/functions/fnc_restoreGear.sqf

```diff
@@ -44,7 +44,7 @@ if (!isNil "_activeWeaponAndMuzzle") then {
         private _index = 0;
 
         while {
-            _index < 100 && {currentWeaponMode _unit != _activeWeaponMode}
+            _index < 299 && {currentWeaponMode _unit != _activeWeaponMode}
```

Comment:
Nothing special about the specific number, just needs to be bigger than biggest index.
This way they all use the same number `299`